internal/coreinternal/attraction/attraction.go

```diff
@@ -305,14 +312,49 @@ func (ap *AttrProc) Process(ctx context.Context, logger *zap.Logger, attrs pcomm
 		case EXTRACT:
 			extractAttributes(action, attrs)
 		case CONVERT:
-			convertAttribute(logger, action, attrs)
+			ap.convertAttribute(action, attrs)
 		}
 	}
 }
 
-func getAttributeValueFromContext(ctx context.Context, key string) (pcommon.Value, bool) {
+func (ap *AttrProc) getAttributeValueFromContext(ctx context.Context, key string) (pcommon.Value, bool) {
+	const (
+		metadataPrefix = "metadata."
+		authPrefix     = "auth."
+	)
+
 	ci := client.FromContext(ctx)
-	vals := ci.Metadata.Get(key)
+	var vals []string
+
+	switch {
+	case strings.HasPrefix(key, metadataPrefix):
+		mdKey := strings.TrimPrefix(key, metadataPrefix)
+		vals = ci.Metadata.Get(mdKey)
+	case strings.HasPrefix(key, authPrefix):
+		if ci.Auth == nil {
+			return pcommon.Value{}, false
+		}
+
+		attrName := strings.TrimPrefix(key, authPrefix)
+		attr := ci.Auth.GetAttribute(attrName)
+
+		switch a := attr.(type) {
+		case string:
+			return pcommon.NewValueString(a), true
+		case []string:
+			vals = a
+		default:
+			ap.logger.Warn(
```

Comment:
This could potentially generate a lot of log lines. We've previously recommended using (to the extent of copy/pasting) [sampled logger](https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/queued_retry.go#L61-L78) though with so many cases it's relevant perhaps we just put it into some common helper package

Reply:
Thanks for the tip, I missed that previously.